File: hw5/q3cpp/Stack.cpp
//Jacob Vargo

#include <vector>
using namespace std;

template<class T, int size>
Stack<T, size>::Stack(int maxsize){
	vec.reserve(maxsize);
}

template<class T, int size>
bool Stack<T, size>::push(T value){
	if (vec.size() == size) return false;
	vec.push_back(value);
	return true;
}

template<class T, int size>
T Stack<T, size>::pop(){
	T rv = vec.back();
	vec.pop_back();
	return rv;
}

template<class T, int size>
bool Stack<T, size>::isEmpty(){
	if (vec.empty()) return true;
	return false;
}

File: hw5/q3java/stack/Stack.java
//Jacob Vargo
package stack;

import java.util.*;

public class Stack<T>{
	protected ArrayList<T> al;
	protected int maxsize;

	public Stack(){
		al = new ArrayList<T>();
		maxsize = 100;
		al.ensureCapacity(maxsize);
	}
	public Stack(int size){
		al = new ArrayList<T>();
		maxsize = size;
		al.ensureCapacity(maxsize);
	}
	public boolean push(T value){
		if (al.size() == maxsize) return false;
		al.add(value);
		return true;
	}
	public T pop(){
		return al.remove(al.size()-1);
	}
	public boolean isEmpty(){
		if (al.isEmpty()) return true;
		return false;
	}
};
